Misalignment updates FST and Simplified FST Geometry

klendathu2k opened the pull request:
Updates to misalignment tables (names, correct timestamps, etc...) for the FST.  Implementation of simplified FST geometry model for track reconstruction.  Support for misaligned geometries in g2t_volume_id.

klendathu2k committed:
FST misaligned geometry initial commit

klendathu2k committed:
BFC support for dev2022m geometry

gavin-wilks committed:
Updating files to work with flattened FST geometry for misalignment

gavin-wilks committed:
Use surveyed fst disk positions in all geometries

klendathu2k committed:
Merge pull request #8 from gavin-wilks/star-agml-fst-misalignment

Star agml fst misalignment

klendathu2k committed:
Resolving some differences with main branch before PR.

klendathu2k committed:
Merge branch 'main' of https://github.com/star-bnl/star-sw into star-bnl-main

klendathu2k committed:
Merge branch 'star-bnl-main' into star-agml-fst-misalignment

genevb reviewed (commented):
I don't have much to speak of on any of this PR (I'm sure misalignment will be useful). A tangential question though: what else is needed for `y2022a`?

klendathu2k commented:
Depends on whether forward tracking is to be supported with y2022a.  If 
we limit
ourselves to support central tracking with the y2022a tag, I can define 
that tomorrow.
Otherwise, I have to double check where we are on the sTGC.

On 2022-01-25 11:47, Gene Van Buren wrote:
> @genevb commented on this pull request.
> 
> I don't have much to speak of on any of this PR (I'm sure misalignment
> will be useful). A tangential question though: what else is needed for
> y2022a?
> 
> --
> Reply to this email directly, view it on GitHub [1], or unsubscribe
> [2].
> Triage notifications on the go with GitHub Mobile for iOS [3] or
> Android [4].
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
> 
> Links:
> ------
> [1] 
> https://github.com/star-bnl/star-sw/pull/285#pullrequestreview-862578744
> [2] 
> https://github.com/notifications/unsubscribe-auth/ANL4LVDWJXH3TCTLARJY2OLUX3H2ZANCNFSM5MYUZBBA
> [3] 
> https://urldefense.com/v3/__https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&amp;mt=8&amp;pt=524675__;!!P4SdNyxKAPE!UHl1kHJRHTQwC75YtuA2_OPEtj8yG6itdC5S16SG_RnlBfdoMWwuSGpZ2ZxkSF8$
> [4] 
> https://urldefense.com/v3/__https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign*3Dnotification-email*26utm_medium*3Demail*26utm_source*3Dgithub__;JSUlJSU!!P4SdNyxKAPE!UHl1kHJRHTQwC75YtuA2_OPEtj8yG6itdC5S16SG_RnlBfdoMWwuSGpZ37x8nhM$

genevb commented:
> Depends on whether forward tracking is to be supported with y2022a. If we limit ourselves to support central tracking with the y2022a tag, I can define that tomorrow. 

If you define y2022a now (sans forward tracking), is there any difference from y2022 (initial)?

klendathu2k commented:
There should not be.  It will be a straight y2022 --> y2022a copy.  
There have
been no updates outside of the forward detectors.

On 2022-01-25 13:21, Gene Van Buren wrote:
>> Depends on whether forward tracking is to be supported with y2022a.
>> If we limit ourselves to support central tracking with the y2022a
>> tag, I can define that tomorrow.
> 
> If you define y2022a now (sans forward tracking), is there any
> difference from y2022 (initial)?
> 
> --
> Reply to this email directly, view it on GitHub [1], or unsubscribe
> [2].
> Triage notifications on the go with GitHub Mobile for iOS [3] or
> Android [4].
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
> 
> Links:
> ------
> [1] 
> https://github.com/star-bnl/star-sw/pull/285#issuecomment-1021478658
> [2] 
> https://github.com/notifications/unsubscribe-auth/ANL4LVAZR4F6S7V3CUCS47LUX3S2ZANCNFSM5MYUZBBA
> [3] 
> https://urldefense.com/v3/__https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&amp;mt=8&amp;pt=524675__;!!P4SdNyxKAPE!UPX-TW3rOjXRMwSI6WTSz5mQ1r9-gK_akkqoTxmcq9y_xjtiXs8o3-5C_R2Qe3k$
> [4] 
> https://urldefense.com/v3/__https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign*3Dnotification-email*26utm_medium*3Demail*26utm_source*3Dgithub__;JSUlJSU!!P4SdNyxKAPE!UPX-TW3rOjXRMwSI6WTSz5mQ1r9-gK_akkqoTxmcq9y_xjtiXs8o3-5CUycmYm4$

genevb commented:
> There should not be. It will be a straight y2022 --> y2022a copy. There have been no updates outside of the forward detectors.

Thanks for the reply. In that case, I am not pressing for freezing y2022a now.

plexoos commented:
Should we store the misalignment values from StarDb/Geometry/fst/*.C in the database instead?

klendathu2k commented:
@plexoos The misalignments are not expected to vary over the course of the run, so we only expect one entry once they have been measured.   The practice for HFT misalignments, and I believe TPC and others as well, has been to store them StarDb area.

klendathu2k commented:
@genevb @perevbnlgov  This geometry update is stalled waiting for a code owner to approve.  Could one of you have a look?  Thanks.

genevb commented on StRoot/StChain/GeometryDbAliases.h:
```diff
@@ -155,12 +155,15 @@ static const DbAlias_t fDbAlias[] = {// geometry  Comment            old
 
-  {"dev2021",     21201210,     1, "dev2021",  "-deprecated- geometry for 2021+ forward program,AgML,xgeometry"},  
+  {"dev2021",     21201210,     1, "dev2021",  "-deprecated- geometry for 2021+ forward program,AgML,xgeometry"},
   {"dev2022",     21211210,     1, "dev2022",  "development geometry for 2022+ forward program,AgML,xgeometry"},
```
Why not use 20211015, as is the case for y2022, for all the dev2022* timestamps?

klendathu2k replied:
I have been trying to keep the R&D / development geometries well advanced in time, so as not to accidentally run them in a production context (thus the +100 years).

genevb commented on StRoot/StChain/GeometryDbAliases.h:
```diff
@@ -155,12 +155,15 @@ static const DbAlias_t fDbAlias[] = {// geometry  Comment            old
   {"dev2022",     21211210,     1, "dev2022",  "development geometry for 2022+ forward program,AgML,xgeometry"},
 
+  {"dev2022m",     20221210,     2, "dev2022m",  "development geometry for 2022+ forward program,AgML,xgeometry"},
+  {"dev2022sm",     20221210,     3, "dev2022sm",  "development geometry for 2022+ forward program,AgML,xgeometry"},
```
Can the comments here indicate the difference between `dev2022`,` dev2022m`, and `dev2022sm`?

klendathu2k replied:
@genevb Comments updated.  Any other issues?

genevb commented:
> The misalignments are not expected to vary over the course of the run, so we only expect one entry once they have been measured. The practice for HFT misalignments, and I believe TPC and others as well, has been to store them StarDb area.

For the TPC, I think surveys and ideal geometries have been stored as CINT files, but the misalignments used in production are stored in the database. If my memory serves me correctly, I think for the HFT that the CINT files were generated from the database tables, but I could be wrong.

-Gene

klendathu2k committed:
Update GeometryDbAliases.h

Comments on geometry tags updated.  Note that tags are documented in the StarVMC/Geometry/StarGeo.xml steering file as well.

klendathu2k commented:
Looks like trigger simulator is breaking tests?

genevb commented:
> Looks like trigger simulator is breaking tests?

I have created an issue: #303 .

-Gene

veprbl commented:
FYI I've restarted CI after #304 was merged, the tests are green again

starsdong commented:
Are we good with this one? @jdbrice or @perevbnlgov need your approval to proceed.

plexoos reviewed (approved):
I trust this been tested and ready

starsdong commented:
Dmitri, let us move ahead and merge this in then. Thanks
